neurom/morphmath.py

```diff
@@ -81,6 +81,34 @@ def f(a, b, c):
     return f(r2, r1, 1. - fraction) if r1 > r2 else f(r1, r2, fraction)
 
 
+def path_fraction_id_offset(points, fraction, relative_offset=False):
+    '''Find the segment which corresponds to the fraction
+    of the path length along the piecewise linear curve which
+    is constructed from the set of points.
+
+    Args:
+        points: an iterable of indexable objects with indices
+        0, 1, 2 correspoding to 3D cartesian coordinates
+        fraction: path length fraction (0.0 <= fraction <= 1.0)
+        relative_offset: return absolute or relative segment distance
+
+    Returns:
+        (segment ID, segment offset) pair.
+    '''
+    if not (0. <= fraction <= 1.0):
+        raise ValueError("Invalid fraction: %.3f" % fraction)
+    pts = np.array(points)[:, COLS.XYZ]
+    lengths = np.linalg.norm(np.diff(pts, axis=0), axis=1)
```

Comment:
I guess I'm supposed to use `point_dist` for calculating segment lengths, but it feels rather clumsy when I need all the lengths at once; hopefully reusing COLS.XYZ range alleviates this deviation.
In fact, I would argue that `point_dist` itself should be reusing NumPy `diff + norm`, but that's not relevant to this change.